pkg/login/client_login.go

```diff
@@ -155,6 +155,39 @@ func GuestLogin(config *config.Config) (string, error) {
 	return guest_user.Url, nil
 }
 
+func CILogin(config *config.Config, apiKey string) error {
+	parsedBaseURL, err := url.Parse(config.APIBaseURL)
+	if err != nil {
+		return err
+	}
+
+	client := &hookdeck.Client{
+		BaseURL: parsedBaseURL,
+		APIKey: apiKey,
+	}
+
+	response, err := client.CreateCIClient(hookdeck.CreateCIClientInput{
+		DeviceName: config.Profile.DeviceName,
+	})
+	if err != nil {
+		return err
+	}
+
+	validateErr := validators.APIKey(response.APIKey)
+	if validateErr != nil {
+		return validateErr
+	}
+
+	config.Profile.APIKey = response.APIKey
+	config.Profile.ClientID = response.ClientID
+	config.Profile.DisplayName = response.UserName
+	config.Profile.TeamName = response.TeamName
+	config.Profile.TeamMode = response.TeamMode
+	config.Profile.TeamID = response.TeamID
```

Comment:
Even though this is my approach, I feel a tad uneasy about this. We don't really want to persist these credentials as they're supposed to be short-lived. However, there's no guarantee that we won't attempt to persist these data to config at a later time.

Should I be adding some mechanism to prevent that from happening?

Reply:
My first thought is that this is an ephemeral context and therefore doesn't really matter. However, thinking more about this makes me think that we should actually lean into this. The "downside" to `--ci` is that you build that auth context on that specific command and can't use multiple command, terminate the listen and rerun with the same context etc.

This brings me to think that having a `hookdeck ci [api-key]` command, which creates the client and writes the credential that is then used in `hookdeck listen` (no --ci) might make more sense. Basically `ci` would be an alternative command to `login` which authenticates you for future commands with different context.

Therefore in your CI pipeline you would

```
hookdeck ci $apikey
hookdeck listen 9001 [...]
```

What do you think?

Reply:
My first thought is that this is an ephemeral context and therefore doesn't really matter. However, thinking more about this makes me realize that we should actually lean into this. The "downside" to `--ci` is that you build that auth context on that specific command and can't use multiple command, terminate the listen and rerun with the same context etc.

This brings me to think that having a `hookdeck ci [api-key]` command, which creates the client and writes the credential that is then used in `hookdeck listen` (no --ci) might make more sense. Basically `ci` would be an alternative command to `login` which authenticates you for future commands with different context.

Therefore in your CI pipeline you would

```
hookdeck ci $apikey
hookdeck listen 9001 [...]
```

What do you think?

Reply:
Sending an other curve ball here, why can't we just do `hookdeck login --api-key $` which would then call `APIKeyLogin` and `createAPIKeyClient`. I'm pretty sure there will be other usecases for using the CLI that way other then in a CI.

Reply:
Fair point, it's functionally the same, and then it's more a DX question. Does having 2 "login", one without a user and one with deserve different commands for clarity? `hookdeck ci` is very explicit about the use case for user-less login?

Reply:
You making a good point too, either works honestly! 

One last curve ball, what if we instead got the API key directly from the env variables to skip the need for this command.

So the CLI have 3 ways for authenticating in this order:

1. Read from a config file to get the API key
2. Read from the env to get the key
3. Current login

The two first steps would be similar to how the AWS CLI works which always made sense to me.

We would still need to add a command to the CLI, something like `hookdeck configure` which would ask for the API key to save in a file. We could even remove then `hookdeck login` when we introduce per user keys.

Reply:
> One last curve ball, what if we instead got the API key directly from the env variables to skip the need for this command.

Yes, that's in the design too, just haven't implemented it yet. Will add that as well.

> 1. Read from a config file to get the API key

I'm not so sure about this. We don't want to store secrets in config file, and if this is running in CI then I don't think this is that helpful compared to reading straight from env.

> We could even remove then hookdeck login when we introduce per user keys.

Is this something we're doing at some point? Just curious, haven't heard of this yet but I do think it may be helpful too.

Reply:
Back to the `hookdeck login --api-key` vs `hookdeck ci --api-key`, it's probably just DX. I like the CI command since it's very specific and clear on its usage. If we find ourselves wanting to do support the other approach too then we can add a command (or alias or something) for that.

Reply:
> Yes, that's in the design too, just haven't implemented it yet. Will add that as well.

Sweet!

> I'm not so sure about this. We don't want to store secrets in config file, and if this is running in CI then I don't think this is that helpful compared to reading straight from env.

That would be to use on your machine, for example, to replace the login and not in your CI, I would imagine that in the context of a CI, you would always rely on the env variable. So I don't see any inconvenient saving secrets on a dev's computer, it would be just like saving an ssh key or something else. Anyhow, this can always be added at a later point. 

For reference: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-quickstart.html

> Is this something we're doing at some point? Just curious, haven't heard of this yet but I do think it may be helpful too.

It has been discussed in the past and is probably something we'll do eventually, but not a priority right now.

Reply:
> Back to the `hookdeck login --api-key` vs `hookdeck ci --api-key`, it's probably just DX. I like the CI command since it's very specific and clear on its usage. If we find ourselves wanting to do support the other approach too then we can add a command (or alias or something) for that.

Sounds good!

Reply:
If we'll read credentials from a file or ENV then having different commands is important I think. Otherwise it becomes unclear what the command will depending on what the credentials in the ENV are.